### PR TITLE
Wandb sweeps

### PR DESCRIPTION
**Pull request by jhaux**
(Could you merge my #253 pull request first?)

The great functionality of wandb sweeps cannot be used out of the box
with edflow. The new `edprep` command helps with this. Now you can
1. Define a sweep from you wandb client
2. edprep the sweep.yaml so that it is compatible with edflow
3. run the wandb sweep
For more details the the docstring in edprep

There are two changes that maybe need discussion:
1. I changed the behavior of the Template Iterator from setting the wandb project id to setdefaulting it. -> I am not sure if this introduces problems, but I have not run into any. See line 75 in edflow/iterators/template_iterator.py
2. I moved the argument parser out of the edflow script to make it reusable and changed the type of the action arguments (`-t, -d`) to a custom type, which should yield the same behavior as before but also allow for the explicit passing of the boolean arguments in various forms. This is necessary, as wandb needs to get all arguments passed to edflow in a specific form, after which it will pass them explicitly as `--key=value` pairs. I have not encountered problems with this new handling, but would like to make sure that a second brain wraps itself around this problem.

### CODE REVIEW

File: edflow/iterators/template_iterator.py [hunk 1]
```diff
@@ -72,8 +72,8 @@ def __init__(self, *args, **kwargs):
                 )
 
                 os.environ["WANDB_RESUME"] = "allow"
-                os.environ["WANDB_RUN_ID"] = ProjectManager.root.strip("/").replace(
-                    "/", "-"
+                os.environ.setdefault(
+                    "WANDB_RUN_ID", ProjectManager.root.strip("/").replace("/", "-")
```
Comment (pesser) [inline edflow/iterators/template_iterator.py:76]: definitely better

File: edflow/args.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import argparse
+
+
+def get_parser(**parser_kwargs):
+    def str2bool(v):
```
Comment (pesser) [inline edflow/args.py:5]: i guess it is fine like this, but in `parse_unknown_args` we use `ast.literal_eval` to parse values. for consistency we could do the same here, but `True` and `False` are covered anyway. Both are fine by me, just let me know if you want to keep it this way or change it before I merge

**Review by pesser — CHANGES_REQUESTED** [review]
Looks great, I'm looking forward to trying it! Let me know about my comment and we can merge

File: edflow/args.py (hunk 2 above)
Comment (jhaux) [reply edflow/args.py:5]: As discussen in #255 and #231 we should aim for unified behavior, i.e. I would like to change it. That said, once we implement the unified approach (in #231 or elswhere) this should get automatically fixed. Thus I would opt for merging like this.

**Review by pesser — APPROVED** [review]

**Pull request merged**